bracket-push.json

```diff
@@ -0,0 +1,49 @@
+{
+  "cases": [
+    {
+      "description": "empty string",
+      "input": "''",
+      "expected": true
+    },
+    {
+      "description": "unpaired brackets",
+      "input": "[[",
+      "expected": false
+    },
+    {
+      "description": "wrong ordered brackets",
+      "input": "}{",
+      "expected": false
+    },
+    {
+      "description": "paired and nested brackets",
+      "input": "([{}({}[])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested brackets",
+      "input": "([{])",
+      "expected": false
+    },
+    {
+      "description": "paired and wrong nested brackets",
+      "input": "[({]})",
+      "expected": false
+    },
+    {
+      "description": "paired with whitespace",
+      "input": "{ }",
+      "expected": true
+    },
+    {
+      "description": "paired and nested with non-bracket chars",
+      "input": "([{f00: bar}({}[baz])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested with non-bracket chars",
+      "input": "([{this: sh0ld({}[be] false)])",
```

Comment:
Any reason the 0 should be in here or in "foo" above?  Or is it just distraction?  Nothing in the test title is telling me what the zeros are for, what problem they are on exhibit for.